### PR TITLE
Exposing RSA-PSS

### PR DESCRIPTION
**Pull request by Frosne**
This PR aims in exposing the RSA-PSS functionality for Mozilla.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -928,9 +928,9 @@ dist/mozilla/Makefile.basic: HASH_BUNDLE = -bundle Hacl.Hash.*,Hacl.HKDF,Hacl.HM
 dist/mozilla/Makefile.basic: HPKE_BUNDLE = -bundle 'Hacl.HPKE.*'
 dist/mozilla/Makefile.basic: P256_BUNDLE= -bundle Hacl.P256,Hacl.Impl.ECDSA.*,Hacl.Impl.SolinasReduction,Hacl.Impl.P256.*
 dist/mozilla/Makefile.basic: K256_BUNDLE= -bundle Hacl.K256.ECDSA,Hacl.Impl.K256.*,Hacl.K256.*,Hacl.EC.K256
-dist/mozilla/Makefile.basic: RSAPSS_BUNDLE = -bundle Hacl.Impl.RSAPSS.*,Hacl.Impl.RSAPSS,Hacl.RSAPSS
+# dist/mozilla/Makefile.basic: RSAPSS_BUNDLE = -bundle Hacl.Impl.RSAPSS.*,Hacl.Impl.RSAPSS,Hacl.RSAPSS
 dist/mozilla/Makefile.basic: FFDHE_BUNDLE = -bundle Hacl.Impl.FFDHE.*,Hacl.Impl.FFDHE,Hacl.FFDHE
-dist/mozilla/Makefile.basic: BIGNUM_BUNDLE = -bundle Hacl.Bignum256_32,Hacl.Bignum4096_32,Hacl.Bignum256,Hacl.Bignum4096,Hacl.Bignum32,Hacl.Bignum64,Hacl.GenericField32,Hacl.GenericField64,Hacl.Bignum.*,Hacl.Bignum
+dist/mozilla/Makefile.basic: BIGNUM_BUNDLE = -bundle Hacl.Bignum=Hacl.Bignum,Hacl.Bignum.* -bundle Hacl.Bignum256_32,Hacl.Bignum4096_32,Hacl.Bignum256,Hacl.Bignum4096,Hacl.Bignum32,Hacl.Bignum64,Hacl.GenericField32,Hacl.GenericField64,Hacl.Bignum.*,Hacl.Bignum
```
Comment (msprotz) [inline Makefile:933]: no need to repeat Hacl.Bignum on the right-hand side of the equals sign

**Review by msprotz — CHANGES_REQUESTED** [review]
Please make sure your dist/ diff is clean, right now, I think the file Hacl_Hash_... is superfluous. Other than that, looks good.

**Comment by Frosne** [comment]
I think Hacl_Hash* is expected, RSA PSS uses a hash function.

**Comment by msprotz** [comment]
Then maybe append `[rename=Hacl_Hash]` to your HASH_BUNDLE so that the filename is more palatable?

**Comment by msprotz** [comment]
- Please resolve conflicts
- Please refresh dist/mozilla -- right now I still see that very long filename in there.

```
git rm -rf dist/mozilla/*
make compile-mozilla
git add dist/mozilla
git commit -m "Refresh dist/mozilla"
```

**Comment by msprotz** [comment]
The build log contains errors about 

```
make[2]: *** [<builtin>: Hacl_Bignum256_32_Hacl_Bignum4096_32_Hacl_Bignum256_Hacl_Bignum4096_Hacl_Bignum32_Hacl_Bignum64_Hacl_GenericField32_Hacl_GenericField64_Hacl_Bignum_Hacl_Bignum.o] Error 1
```

so this still doesn't appear to be working. Let me try to suggest an alternative today.

**Comment by msprotz** [comment]
New error:
```
internal/Hacl_Bignum.h:39:10: fatal error: lib_intrinsics.h: No such file or directory
   39 | #include "lib_intrinsics.h"
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
Makefile.basic:53: Hacl_Chacha20Poly1305_256.d: No such file or directory
make[2]: *** [Makefile.basic:48: Hacl_RSAPSS.d] Error 1
```

**Comment by msprotz** [comment]
Once we merge #649, we can address this very simply.

**Pull request closed without merging**